medusa/providers/torrent/json/torrentleech.py

```diff
@@ -0,0 +1,246 @@
+
+        # Continue only if at least one release is found
+        if not torrents:
+            log.debug('Data returned from provider does not contain any torrents')
+            return items
+
+        for torrent in torrents:
+            try:
+                title = torrent['name']
+                download_url = self.urls['download'].format(id=torrent['fid'], file=torrent['filename'])
+
+                seeders = int(torrent['seeders'])
+                leechers = int(torrent['leechers'])
+
+                # Filter unseeded torrent
+                if seeders < min(self.minseed, 1):
+                    if mode != 'RSS':
+                        log.debug("Discarding torrent because it doesn't meet the"
+                                  " minimum seeders: {0}. Seeders: {1}", title, seeders)
+                    continue
+
+                size = convert_size(torrent['size']) or -1
+
+                pubdate_raw = torrent['addedTimestamp']
+                pubdate = self.parse_pubdate(pubdate_raw, timezone=user_timezone)
+
+                item = {
+                    'title': title,
+                    'link': download_url,
+                    'size': size,
+                    'seeders': seeders,
+                    'leechers': leechers,
+                    'pubdate': pubdate,
+                }
+                if mode != 'RSS':
+                    log.debug('Found result: {0} with {1} seeders and {2} leechers',
+                              title, seeders, leechers)
+
+                items.append(item)
+            except (AttributeError, TypeError, KeyError, ValueError, IndexError):
+                log.exception('Failed parsing provider.')
+
+        return items
+
+    def login(self):
+        """Login method used for logging in before doing search and torrent downloads."""
+        cookies = dict_from_cookiejar(self.session.cookies)
+        if any(cookies.values()) and cookies.get('member_id'):
+            return True
+
+        login_params = {
+            'username': self.username,
+            'password': self.password,
+            'login': 'submit',
+            'remember_me': 'on',
+        }
+
+        response = self.session.post(self.urls['login'], data=login_params)
+        if not response or not response.text:
+            log.warning('Unable to connect to provider')
+            return False
+
+        if '<title>Login :: TorrentLeech.org</title>' in response.text:
+            log.warning('Invalid username or password. Check your settings')
+            return False
+
+        return True
+
+    def _pagination(self, data, mode, search_url):
+        """If needed, query the next page(s) of results, parse them and return."""
+        results = []
+        num_found = data.get('numFound', 0)
+        per_page = data.get('perPage', 35)
+
+        if per_page < 100 and num_found > per_page:
+            log.info('It is recommended to change "Default Results Per Page" to 100'
```

Comment:
@medariox Do you think this should be a warning?
I would mean less calls to TorrentLeech, which is good for the reputation...

Reply:
Yep, I think it'd be better.